api/envoy/api/v2/cds.proto

```diff
@@ -657,6 +657,35 @@ message Cluster {
   // :ref:`lb_policy<envoy_api_field_Cluster.lb_policy>` field has the value
   // :ref:`LOAD_BALANCING_POLICY_CONFIG<envoy_api_enum_value_Cluster.LbPolicy.LOAD_BALANCING_POLICY_CONFIG>`.
   LoadBalancingPolicy load_balancing_policy = 41;
+
+  // [#not-implemented-hide:] If present, tells the client where to send load reports via LRS.
+  //
+  // If multiple clusters are configured to use the same LRS server, then the data for all of
+  // those clusters may be sent on the same LRS stream.
+  //
+  // Note that if LRS is configured this way, then there is no need for the LRS server to send
+  // LoadStatsResponse messages on the LRS stream, because the data it would have sent is already
+  // configured here.
+  message LoadReportConfig {
+    // A management server endpoint to stream load stats to via
+    // *StreamLoadStats*. This must have :ref:`api_type
+    // <envoy_api_field_core.ApiConfigSource.api_type>` :ref:`GRPC
+    // <envoy_api_enum_value_core.ApiConfigSource.ApiType.GRPC>`.
+    core.ApiConfigSource lrs_server = 1;
```

Comment:
In discussing this with @vishalpowar, one problem here is that some management server deployments want the LRS stream to be sent to the same management server that is providing the CDS request, but they don't currently know what name or credentials their clients are using to access the management server.  It would be more work for them to figure that out just to give the client information that it already has (because it used the name and credentials to access the CDS request in the first place).

Currently, ConfigSource has a [special case for ADS](https://github.com/envoyproxy/envoy/blob/d5515aed33244952786ea4b7732af9a49b26b6f6/api/envoy/api/v2/core/config_source.proto#L111) that basically says "use the same ADS stream you're already using", but that won't work here, because LRS can't be sent over ADS.  To address this, can we add a new field to the "oneof" in ConfigSource that basically says "talk to the same server you're getting this message from" but without being linked to ADS?

I think this new type of ConfigSource might also be useful in cases that are not using ADS but where LDS points to RDS on the same server or where CDS points to EDS on the same server.

@htuch and @mattklein123, WDYT?

Reply:
This makes sense to me but can we track this in a new issue/proposal?

Reply:
Sure.  I've sent out #8201 for review.